diff -ru a/src/runtime/internal/sys/arch_arm.go b/go/src/runtime/internal/sys/arch_arm.go
--- a/src/runtime/internal/sys/arch_arm.go	2016-07-18 09:24:08.000000000 -0700
+++ b/go/src/runtime/internal/sys/arch_arm.go	2016-11-21 00:22:28.840788782 -0800
@@ -8,7 +8,7 @@
 	TheChar       = '5'
 	BigEndian     = 0
 	CacheLineSize = 32
-	PhysPageSize  = 65536*GoosNacl + 4096*(1-GoosNacl)
+	PhysPageSize  = 16384
 	PCQuantum     = 4
 	Int64Align    = 4
 	HugePageSize  = 0
Only in b/go/src/runtime/internal/sys: arch_arm.go.orig
diff -ru a/src/syscall/syscall_linux_arm.go b/go/src/syscall/syscall_linux_arm.go
--- a/src/syscall/syscall_linux_arm.go	2016-07-18 09:24:08.000000000 -0700
+++ b/go/src/syscall/syscall_linux_arm.go	2016-11-21 00:22:28.840788782 -0800
@@ -11,7 +11,7 @@
 	_SYS_getdents = SYS_GETDENTS64
 )
 
-func Getpagesize() int { return 4096 }
+func Getpagesize() int { return 16384 }
 
 func TimespecToNsec(ts Timespec) int64 { return int64(ts.Sec)*1e9 + int64(ts.Nsec) }
 
@@ -139,8 +139,8 @@
 }
 
 func mmap(addr uintptr, length uintptr, prot int, flags int, fd int, offset int64) (xaddr uintptr, err error) {
-	page := uintptr(offset / 4096)
-	if offset != int64(page)*4096 {
+	page := uintptr(offset / 16384)
+	if offset != int64(page)*16384 {
 		return 0, EINVAL
 	}
 	return mmap2(addr, length, prot, flags, fd, page)
